### ISSUE TITLE
Drop event not delivered on Xfce

### ISSUE THREAD

**Issue by btrap** [glfw/glfw#968]
I am currently working on a computer running Debian 9.0 (stretch) with the XFCE desktop environment.  I have tested many GLFW callbacks and the only one I am seeing issues with is glfwSetDropCallback.  Nothing seems to happen with it, I can't cause anything to trigger the callback.  I have attached test code that works on Windows but not on Debian.  I have also queried X11 XdndAware and it returns true, that is in the test code as well.
```c
// TEST GLFW glfwSetDropCallback
glfwInit();

// You might need some of the window hints

GLFWwindow *window = glfwCreateWindow(600, 400, "Test glfwSetDropCallback", NULL, NULL);

glfwSetDropCallback(window, [](GLFWwindow*, int count, const char **paths)
{
    std::cout << "Drag and Drop count - " << count << std::endl;

    for (unsigned int i = 0; i < count; ++i)
        std::cout << "- " << paths[i] << std::endl;
});

while (!glfwWindowShouldClose(window))
{
    glfwSwapBuffers(window);
    glfwPollEvents();
}

glfwDestroyWindow(window);

glfwTerminate();

// TEST X11 Xdnd
Display *display =XOpenDisplay((char *)0);
if(XInternAtom(display, "XdndAware", False))
    std::cout << "Supported" << std::endl;
```

**Comment by elmindreda** [comment]
What version of GLFW are you using?

**Comment by MrVallentin** [comment]
The version given by `glfwGetVersionString()` is:

    3.2.1 X11 GLX EGL clock_gettime /dev/js Xf86vm shared

[I referred him from Stack Overflow.](http://stackoverflow.com/questions/42798513/having-issues-with-glfws-callback-glfwsetdropcallback-on-debian/42799574#42799574)

**Comment by btrap** [comment]
Exactly what Mr Vallentin said.

**Comment by elmindreda** [comment]
Unable to reproduce on MATE.

**Comment by elmindreda** [comment]
Verified on Xfce.

**Comment by elmindreda** [comment]
Bug is due to Thunar providing `text/uri-list` while GLFW assumed `UTF8_STRING` was supported. Fix incoming.

**Comment by elmindreda** [comment]
This and many other bugs relating to XDND should be fixed now with 6a65341e14bc7745c52fe86fe53be08dbd682dd9.

**Comment by btrap** [comment]
Thank you

**Comment by cpicanco** [comment]
@elmindreda , what is the version of Thunar supported by glfw?

**Comment by elmindreda** [comment]
@cpicanco Drag and drop should work with any file manager that implements the XDND protocol. Is it not working with a particular version of Thunar?

**Comment by cpicanco** [comment]
Thank you for the information. I must say that I am just a user conserning glfw stuff. So what I can tell you for sure is that the pupil player application that is based on glfw does not receive drops from Xfce Thunar 1.6.3 but but will receive drops from GNOME nautilus 3.14.1. Is there anything else I can do to fix this? (also the title bar is missing)

**Comment by elmindreda** [comment]
@cpicanco Are you using a copy of GLFW that includes the fix above?

**Comment by cpicanco** [comment]
The pupil application uses libglfw3-dev as a dependency.

**Comment by elmindreda** [comment]
The version of GLFW you got from your package system isn't new enough to contain this fix.

**Comment by cpicanco** [comment]
Even in Ubuntu Xenial?

**Comment by elmindreda** [comment]
Ubuntu Xenial has GLFW 3.1.2, which was released October 12, 2015. The fix above will be included in GLFW 3.3.

**Comment by cpicanco** [comment]
Thank you Camilla.
